src/librbd/Journal.cc

```diff
@@ -353,59 +350,28 @@ bool Journal<I>::is_journal_supported(I &image_ctx) {
 
 template <typename I>
 int Journal<I>::create(librados::IoCtx &io_ctx, const std::string &image_id,
-		       uint8_t order, uint8_t splay_width,
-		       const std::string &object_pool, bool non_primary,
+                       uint8_t order, uint8_t splay_width,
+                       const std::string &object_pool, bool non_primary,
                        const std::string &primary_mirror_uuid) {
+  assert(non_primary ^ primary_mirror_uuid.empty());
+
   CephContext *cct = reinterpret_cast<CephContext *>(io_ctx.cct());
   ldout(cct, 5) << __func__ << ": image=" << image_id << dendl;
 
-  librados::Rados rados(io_ctx);
-  int64_t pool_id = -1;
-  if (!object_pool.empty()) {
-    IoCtx data_io_ctx;
-    int r = rados.ioctx_create(object_pool.c_str(), data_io_ctx);
-    if (r != 0) {
-      lderr(cct) << __func__ << ": "
-                 << "failed to create journal: "
-		 << "error opening journal objects pool '" << object_pool
-		 << "': " << cpp_strerror(r) << dendl;
-      return r;
-    }
-    pool_id = data_io_ctx.get_id();
-  }
-
   Journaler journaler(io_ctx, image_id, IMAGE_CLIENT_ID,
                       cct->_conf->rbd_journal_commit_age);
 
-  int r = journaler.create(order, splay_width, pool_id);
-  if (r < 0) {
-    lderr(cct) << __func__ << ": "
-               << "failed to create journal: " << cpp_strerror(r) << dendl;
-    return r;
-  }
-
-  cls::journal::Client client;
-  cls::journal::Tag tag;
   journal::TagData tag_data;
+  tag_data.mirror_uuid = (non_primary ? primary_mirror_uuid
+                          : LOCAL_MIRROR_UUID);
```

Comment:
... actually, would anyone call this static create method with non_primary and primary_mirror_uuid arguments?  That should only be used in the image::CreateRequest path.